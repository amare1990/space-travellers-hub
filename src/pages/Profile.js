import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets } from '../redux/rockets/rockets';
import { fetchMissions } from '../redux/missions/missions';

import '../styles/Profile.css';

const Profile = () => {
  const rockets = useSelector((state) => state.rockets.rockets);
  const missions = useSelector((state) => state.missions.missions);
  const dispatch = useDispatch();
  useEffect(() => {
    if (rockets.length === 0) {
      dispatch(fetchRockets());
    }
    if (missions.length === 0) {
      dispatch(fetchMissions());
    }
  });

  const joinedMissions = missions.filter((mission) => mission.reserved);
  let flagMission = false;
  if (joinedMissions.length !== 0) {
    flagMission = true;
  }

  const reservedRockets = rockets.filter((rocket) => rocket.reserved);
  let flagRocket = false;
  if (reservedRockets.length !== 0) {
    flagRocket = true;
  }

  return (
    <div className="profile">

      <div className="reserved-missions-wrapper">
        <h3 className="profile-title profile-title-mission">
          My Missions
        </h3>

        <div className="reserved-profile reserved-profile-mission ">
          {
          missions.map((mission) => (
            mission.reserved ? (
              <div key={mission.mission_id} className="reserved-item reserved-mission-item">
                {mission.mission_name}
              </div>
            ) : ''
          ))
          }

        </div>

        <p className="flag-text">
          {flagMission ? '' : 'No Missions Selected Yet!'}
        </p>

      </div>

      <div className="reserved-rockets-wrapper">
        <h3 className="profile-title profile-title-rocket">
          My Rockets
        </h3>
        <div className="reserved-profile reserved-profile-rocket ">
          {
          rockets.map((rocket) => (
            rocket.reserved ? (
              <div key={rocket.rocket_id} className="reserved-item reserved-rocket-item">
                {rocket.rocket_name}

              </div>
            ) : ''
          ))
          }

        </div>
        <p className="flag-text">
          { flagRocket ? '' : 'No Rockets Selected Yet!' }
        </p>
      </div>

    </div>
  );
};

export default Profile;
