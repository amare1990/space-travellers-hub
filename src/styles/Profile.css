.profile {
  display: flex;
  align-items: flex-start;
  padding: 0 120px 0 200px;
}

.reserved-rockets-wrapper,
.reserved-missions-wrapper {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  width: 40vw;
  margin-left: 10px;
}

.reserved-profile {
  list-style: none;
  border-top: 1px solid #999;
  border-right: 1px solid #999;
  border-left: 1px solid #999;
  width: 100%;
}

.reserved-item {
  border-bottom: 1px solid #999;
  padding: 7px 0 20px 10px;
}

.reserved-rocket-item {
  width: 39vw;
}

.reserved-profile-rocket {
  width: 100%;
}

.flag-text {
  padding-top: 20px;
  color: blue;
}

.no-joined-reserved {
  padding-top: 10px;
  font-size: 24px;
  color: #459a;
  text-align: center;
}
