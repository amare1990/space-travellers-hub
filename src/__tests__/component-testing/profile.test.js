import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Profile from '../../pages/Profile';
import store from '../../redux/configureStore';

test('Testing if Profile component renders correctly', () => {
  const profileSnapshot = renderer.create(
    <Provider store={store}>
      <Profile />
    </Provider>,
  ).toJSON();
  expect(profileSnapshot).toMatchSnapshot();
});
