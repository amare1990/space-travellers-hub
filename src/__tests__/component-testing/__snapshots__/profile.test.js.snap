// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing if Profile component renders correctly 1`] = `
<div
  className="profile"
>
  <div
    className="reserved-missions-wrapper"
  >
    <h3
      className="profile-title profile-title-mission"
    >
      My Missions
    </h3>
    <div
      className="reserved-profile reserved-profile-mission "
    />
  </div>
  <div
    className="reserved-rockets-wrapper"
  >
    <h3
      className="profile-title profile-title-rocket"
    >
      My Rockets
    </h3>
    <div
      className="reserved-profile reserved-profile-rocket "
    />
  </div>
</div>
`;
